Sources/MessagingPush/PushHandling/AutomaticPushClickHandling.swift

```diff
@@ -0,0 +1,49 @@
+import CioInternalCommon
+import Foundation
+
+/**
+ Sets up the CIO SDK to automatically handle events related to push notifications such as when a push is clicked or deciding if a push should be shown while app is in foreground.
```

Comment:
Because this feature is a bit complex, I thought that it would be good to document in the code how this feature is implemented, the problems we have to solve when making it, and the goals that this feature has. 

Ask questions if you read this and do not understand part of the feature. I hope these docs can be helpful far into the future as we maintain this feature.